autorest/azure/azure.go

```diff
@@ -147,6 +149,40 @@ func IsAzureError(e error) bool {
 	return ok
 }
 
+// Resource contains details about an Azure resource
+type Resource struct {
+	subscription string
+	resourceGroup string
+	provider string
+	resourceType string
+	resourceName string
```

Comment:
Should this ones be exported?

Reply:
what do you mean by these ones ? The type itself needs to be exported, that is what will contain the parsed result.

Reply:
or you want to return an unexported type from the parse function ? I dont think that is a good pattern. Golint even complains about that https://github.com/golang/lint/issues/210. 

That being said if we have a good reason to not export it we can do that.

Reply:
She's asking if the fields should be exported which I think they should yes?  If not then how do you access their values?

Reply:
Oh sorry my bad yes

Reply:
What @jhendrixMSFT is what I meant, yes